src/savegame_json.cpp

```diff
@@ -1560,13 +1559,18 @@ void vehicle_part::deserialize(JsonIn &jsin)
     data.read("target_second_y", target.second.y);
     data.read("target_second_z", target.second.z);
 
+    // legacy items may specify ammo via the deprecated amount field
+    // items tagged VEHICLE are already migrated so we can ignore those
+    int amount;
+    if( data.read( "amount", amount ) ) {
+        if( !base.has_flag( "VEHICLE" ) && ammo_capacity() > 0 ) {
```

Comment:
Saves between battery update PR and this PR have fuel tanks with "VEHICLE" flag, but `amount` rather than contents.